src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs

```diff
@@ -68,6 +68,11 @@ public TestEngine() : this(TestRuntimeProviderManager.Instance, new ProcessHelpe
         requestData.MetricsCollection.Add(
             TelemetryDataConstants.ParallelEnabledDuringDiscovery,
             parallelLevel > 1 ? "True" : "False");
+        requestData.MetricsCollection.Add(
+            TelemetryDataConstants.TestSessionId,
+            discoveryCriteria.TestSessionInfo != null
```

Comment:
nit: `discoveryCriteria.TestSessionInfo?.Id ?? string.Empt`

Reply:
Will fix this in a subsequent PR as I don't want to go through the review cycle again.